common/menu-entry.c

```diff
@@ -322,42 +322,36 @@ bool menuEntryLoad(menuEntry_s* me, const char* name, bool shortcut) {
 }
 
 void menuEntryParseIcon(menuEntry_s* me) {
-    uint8_t *imageptr = NULL;
     size_t imagesize = 256*256*3;
+    unsigned char *imageptr = (unsigned char*)malloc(imagesize);
+
+    if (imageptr == NULL) return;
+
+    me->icon_gfx = (uint8_t*)imageptr;
+    int w,h,samp;
 
     if (me->icon_size==0 || me->icon==NULL) return;
 
-    njInit();
+    tjhandle _jpegDecompressor = tjInitDecompress();
+
+    if(tjDecompressHeader2(_jpegDecompressor, me->icon, me->icon_size, &w, &h, &samp)==-1) return;
+
+    if (w != 256 || h != 256 ) return; //The decoded image must be 256x256.
 
-    if (njDecode(me->icon, me->icon_size) != NJ_OK) {
-        njDone();
+    if(tjDecompress2(_jpegDecompressor, me->icon, me->icon_size, imageptr, w, 0/*pitch*/, h, TJPF_RGB, TJFLAG_ACCURATEDCT)==-1)//The decoded image must be RGB
         return;
-    }
 
     me->icon_size = 0;
     free(me->icon);
     me->icon = NULL;
 
-    if ((njGetWidth() != 256 || njGetHeight() != 256 || (size_t)njGetImageSize() != imagesize) || njIsColor() != 1) {//The decoded image must be RGB and 256x256.
-        njDone();
-        return;
-    }
-
-    imageptr = njGetImage();
-    if (imageptr == NULL) {
-        njDone();
-        return;
-    }
-
     me->icon_gfx = (uint8_t*)malloc(imagesize);
-    if (me->icon_gfx == NULL) {
-        njDone();
-        return;
-    }
 
-    memcpy(me->icon_gfx, imageptr, imagesize);
+    if (me->icon_gfx == NULL) return;
```

Comment:
Get rid of this and just set it to imageptr directly.

Reply:
(starting with the malloc)

Reply:
You cant pass a null ptr into the tjDecompress2 function

Reply:
This is post-decompression ...

Reply:
is it really that bad if we allocated 192kb on the stack and its freed right after the function ends?